### PR TITLE
feat: allow *.xml (beside *.log) files as sample input files

### PR DESCRIPTION
**Pull request by hsekowski-splunk**

### CODE REVIEW

**Comment by arys-splunk** [comment]
I think it is a good one, @nandinivij WDYT?

**Review by arys-splunk — COMMENTED** [review]
What about this line https://github.com/splunk/pytest-splunk-addon/blob/23deab578dad8aed8b62b5edaa85ecb45e8a736e/pytest_splunk_addon/standard_lib/event_ingestors/requirement_event_ingester.py#L111? Do we need to change this?

And I would like to see a docker-based test that confirms that this feature works.

**Comment by nandinivij** [comment]
We will need changes unit test side too. Currently, we generate errors for anything other than .log and do not run prerequisite tests for that.

File: tests/unit/tests_standard_lib/test_requirement_tests/test_test_generator.py [hunk 1]
```diff
@@ -133,11 +133,27 @@ def test_extract_params():
...
             ["syslog"],
-            {"event": ["event_1", "event_2"]},
-            [["model_1:dataset_1", "model_2:dataset_2"], ["model_3:dataset_3"]],
-            ["event_name_2"],
+            {"event": ["<34>Oct 11 22:14:15 machine1 pr1:event_1"]},
+            [["model_1:dataset_1", "model_2:dataset_2"]],
+            ["event_name_1"],
             [{"field1": "value1", "field2": "value2"}, {"field3": "value3"}],
-            [],
+            [
```
Comment (nandinivij) [inline tests/unit/tests_standard_lib/test_requirement_tests/test_test_generator.py:140]: this test does not run on docker. Tests part of test_addon.py does

Comment (hsekowski-splunk) [reply tests/unit/tests_standard_lib/test_requirement_tests/test_test_generator.py:140]: @nandinivij ,
Docker test added
@arys-splunk , FYI

**Review by arys-splunk — APPROVED** [review]

**Pull request merged**